build_wheels.py

```diff
@@ -0,0 +1,227 @@
+import json
+import re
+
+import requests
+from packaging.requirements import InvalidRequirement
+from packaging.requirements import Requirement
+
+import constants
+
+idf_constraints_url = 'https://dl.espressif.com/dl/esp-idf/espidf.constraints.'
```

Comment:
This is constant, so it should be upper-case:
```py
IDF_CONSTRAINTS_URL= 'https://dl.espressif.com/dl/esp-idf/espidf.constraints.'
```
.... and possibly in the file with other your constants (`constants.py`)